docs/ref/csrf.txt

```diff
@@ -136,18 +136,30 @@ and read the token from the DOM with JavaScript:
 .. code-block:: html+django
 
     {% csrf_token %}
-    <script type="text/javascript">
-    // using jQuery
-    var csrftoken = jQuery("[name=csrfmiddlewaretoken]").val();
+    <script>
+    var csrftoken = document.querySelector('[name=csrfmiddlewaretoken]').value;
     </script>
 
 Setting the token on the AJAX request
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Finally, you'll have to actually set the header on your AJAX request, while
-protecting the CSRF token from being sent to other domains using
-`settings.crossDomain <https://api.jquery.com/jQuery.ajax/>`_ in jQuery 1.5.1
-and newer:
+Finally, you'll have to actually set the header on your AJAX request. Using the
+`fetch()
+<https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch>`_
+API:
+
+.. code-block:: javascript
+
+    var request = new Request(/* url */);
+    var headers = new Headers();
+    headers.set('X-CSRFToken', csrftoken);
+    fetch(url, {method: 'POST', headers: headers}).then(function(response) {
+        // ...
+    });
+
+Be sure to protect the CSRF token from being sent to other domains. If you're
+using jQuery 1.5.1 or newer, use `settings.crossDomain
+<https://api.jquery.com/jQuery.ajax/>`_:
```

Comment:
What do we make of this sentence now that jQuery has been removed?

Reply:
I'm personally okay to drop this. I don't think we should be recommending jQuery these days, but I also understand people continue to use it so perhaps we should be pragmatic.

I wasn't sure if that would be too controversial for a "trivial" no-ticket change. I think if you agree with dropping this, a ticket should be created so it has a bit more visibility.

Which way would you like to go?

Reply:
Or, and I'd need to test to know, does `fetch`'s `no-cors` option do what one might hope it should here?

Reply:
Good call. Indeed the `fetch()` API automatically blocks the request. Here is a good article about this: https://javascript.info/fetch-crossorigin

And here is what I see in the Firefox console:

```
var headers = new Headers(); headers.set('X-CSRFToken', 'abc'); fetch('https://google.com', {headers: headers}).then(function(response) { console.log(response); });

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://google.com/. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://google.com/. (Reason: CORS request did not succeed).

TypeError: NetworkError when attempting to fetch resource.
```

So, I'll remove the non-fetch examples.

Reply:
That's not quite what we need. If `'https://google.com'` were `https://evil.com/` setting `Access-Control-Allow-Origin: *` we **would** be able to send the CSRF token.

Testing, setting `fetch(request, {method: 'GET', headers: headers, mode: 'same-origin'})` blocks the cross-domain request before it's flighted. 

I will add that. 

Thanks @jdufresne!

Reply:
Ah, I see. That makes sense. Thanks!